js/main.js

```diff
@@ -102,96 +111,110 @@ renderPins();
+    if (card.offer.price) {
+      // выводим цену в .popup__text--price
+      cardAd.querySelector('.popup__text--price').textContent = card.offer.price + ' ₽/ночь';
+    } else {
+      cardAd.querySelector('.popup__text--price').remove();
+    }
 
-  // выводим время заезда и выезда в .popup__text--time
-  cardAd.querySelector('.popup__text--time').textContent = 'Заезд после ' + card.offer.checkin +
-    ', выезд до ' + card.offer.checkout || 'no value';
+    if (card.offer.type) {
+      // выводим тип жилья в .popup__type
+      cardAd.querySelector('.popup__type').textContent = ACCOMMODATION[card.offer.type];
+    } else {
+      cardAd.querySelector('.popup__type').remove();
+    }
 
-  // выводим все доступные удобства в .popup__features
-  cardAd.querySelector('.popup__features').textContent = createFeature(cardAd, card.offer.features) || 'no value';
+    if (card.offer.rooms && card.offer.guests) {
+      // выводим количество комнат и гостей в .popup__text--capacity
+      cardAd.querySelector('.popup__text--capacity').textContent = card.offer.rooms +
+        ' комнаты для ' + card.offer.guests + ' гостей';
+    } else {
+      cardAd.querySelector('.popup__text--capacity').remove();
+    }
 
-  // выводим описание объекта недвижимости в .popup__description
-  cardAd.querySelector('.popup__description').textContent = card.offer.description || 'no value';
+    if (card.offer.checkin && card.offer.checkout) {
+      // выводим время заезда и выезда в .popup__text--time
+      cardAd.querySelector('.popup__text--time').textContent = 'Заезд после ' + card.offer.checkin +
+        ', выезд до ' + card.offer.checkout;
+    } else {
+      cardAd.querySelector('.popup__text--time').remove();
+    }
 
-  // выводим фотографии в .popup__photos
-  cardAd.querySelector('.popup__photo').textContent = createPhoto(cardAd, card.offer.photos) || 'no value';
+    if (card.offer.features) {
+      // выводим все доступные удобства в .popup__features
+      cardAd.querySelector('.popup__features').innerHTML = '';
+      cardAd.querySelector('.popup__features').appendChild(createFeatureFragment(card));
+    } else {
+      cardAd.querySelector('.popup__features').remove();
+    }
 
-  var childElements = cardAd.querySelectorAll(':scope > *');
-  childElements.forEach(function (element) {
-    if (element.textContent === 'no value') {
-      element.classList.add('visually-hidden');
+    if (card.offer.description) {
+      // выводим описание объекта недвижимости в .popup__description
+      cardAd.querySelector('.popup__description').textContent = card.offer.description;
+    } else {
+      cardAd.querySelector('.popup__description').remove();
+    }
+    if (card.offer.photos) {
+      // выводим фотографии в .popup__photos
+      cardAd.querySelector('.popup__photos').removeChild(cardAd.querySelector('.popup__photo'));
+      cardAd.querySelector('.popup__photos').appendChild(createPhotosFragment(card));
```

Comment:
можно сразу заменить содержимое
```suggestion
      cardAd.querySelector('.popup__photos').innerHTML(createPhotosFragment(card));
```